sys/include/net/l2filter.h

```diff
@@ -85,6 +85,7 @@ typedef struct {
  *
  * @return  0 on success
  * @return  -ENOMEM if no empty slot left in list
+ * @return  -EINVAL if addr_len > CONFIG_L2FILTER_ADDR_MAXLEN
```

Comment:
```suggestion
 * @retval 0 on success
 * @retval -ENOMEM if no empty slot left in list
 * @retval -EINVAL if addr_len > CONFIG_L2FILTER_ADDR_MAXLEN
```

Out of scope for this PR, but while you're at it 😇

Reply:
Yes, I agree with the use of `@reval`. I do not agree with the `-EINVAL if addr_len > CONFIG_L2FILTER_ADDR_MAXLEN`. That should be something like

```
 * @retval  -EINVAL `addr_len > CONFIG_L2FILTER_ADDR_MAXLEN` **AND** `NDEBUG` is enabled.
 *                  (Without `NDEBUG`, this causes an `assert()` to blow instead.)
```

Reply:
> This was just a fix for Doxygen, because only one @return is allowed per function. For multiple values you'll have to use @retval.

I fixed that.

> IMO we should add a @pre to document that addr_len <= CONFIG_L2FILTER_ADDR_MAXLEN as a precondition

That is already the case, see line 84.

> we should add [..] state that -EINVAL is only returned if NDEBUG is enabled

Hm, yeah fine. I don't like your suggestion wording tho. Will think about it for a minute.